ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/ProtocolScheduleBuilder.java

```diff
@@ -13,20 +13,24 @@
 package tech.pegasys.pantheon.ethereum.mainnet;
 
 import tech.pegasys.pantheon.config.GenesisConfigOptions;
+import tech.pegasys.pantheon.metrics.MetricsSystem;
 
 import java.util.OptionalLong;
 import java.util.function.Function;
 
 public class ProtocolScheduleBuilder<C> {
 
+  private final MetricsSystem metricsSystem;
   private final GenesisConfigOptions config;
   private final Function<ProtocolSpecBuilder<Void>, ProtocolSpecBuilder<C>> protocolSpecAdapter;
   private final int defaultChainId;
 
   public ProtocolScheduleBuilder(
+      final MetricsSystem metricsSystem,
```

Comment:
Giant constructors don't feel like the Builder pattern to me.  Could this (here or another PR) be re-written to follow the builder pattern?

Reply:
I suspect this should actually be named Factory not Builder so have renamed it as such.  The point is that it encapsulates the logic for interpreting a genesis config and setting up all the protocol specs at the right block numbers.  It's not designed to give the pseudo-named-parameter style you're expecting.

Reply:
Naming it factory is a viable alternative.